cargo-fix/tests/all/edition_upgrade.rs

```diff
@@ -0,0 +1,46 @@
+//! Test that we can use cargo-fix to upgrade our code to work with the 2018
+//! edition.
+//!
+//! We'll trigger the `absolute_path_starting_with_module` lint which should
+//! transform a `use ::foo;` where `foo` is local to `use crate::foo;`.
+
+use super::project;
+
+#[test]
+fn prepare_for_2018() {
+    let p = project()
+        .file(
+            "src/lib.rs",
+            r#"
+                #![allow(unused)]
+                #![feature(crate_in_paths)]
```

Comment:
Oh gee this is gonna make for a not-so-great user experience :(

If users *don't* have this enabled then rustfix will accidentally deduce that its fixes broke your crate, so they'll all be backed out. That's a bummer...